src/solvers/README.md

```diff
@@ -228,7 +228,44 @@ To be documented.
 
 \subsection sat-solving-api-section SAT solving API
 
-To be documented.
+The basic SAT solver interface is in [cnf](\ref cnft). This inherits
+from the [propositional logic decision procedure wrapper](\ref propt).
+
+The interface supports the following operations by default:
+1. Boolean operations on literals (like `and`, `or`, `xor`), etc.
+These take as input two [literal](\ref literalt) and return as
+output another [literal](\ref literalt).
+2. Generating new [literals](\ref literalt) and adding them
+as variables to the formula and returning the number of variables
+or clauses the solver is operating with (with `no_variables`).
+3. Simulate gate operations on literals, applying Tseitin's
+transformation on them. Tseitin's transformation converts a
```

Comment:
Ah, I thought 1. was describing the same?

Reply:
It's weird if I'm honest, I don't know how to describe it better. At least from the preexisting function documentation, the `gate_x` functions seem to be implementing tseitin's transformation, and the `literal_x` functions are just setup functions that end up calling equivalent the `gate_x` ones. Both are part of the interface of `cnft`, and I don't know which ones are to be favoured (they seem to be doing the same thing, but with a different interface).

I'm open to suggestions on how to rephrase it in a way that makes more sense.

Reply:
I believe the `gate_*` simply should not be part of the public interface, AFAIK they aren't actually used outside the class! They are concrete implementations of the `lx` functions, which the `propt` interface prescribes. Would you mind adding a commit to this PR that shifts the `protected` up to line 45 of `cnf.h`?